src/object_store.cpp

```diff
@@ -96,19 +96,35 @@ void add_index(Table& table, size_t col)
     }
 }
 
+DataType to_core_type(PropertyType type)
+{
+    switch (type & ~PropertyType::Flags) {
+        case PropertyType::Int:    return type_Int;
+        case PropertyType::Bool:   return type_Bool;
+        case PropertyType::Float:  return type_Float;
+        case PropertyType::Double: return type_Double;
+        case PropertyType::String: return type_String;
+        case PropertyType::Date:   return type_Timestamp;
+        case PropertyType::Data:   return type_Binary;
+        case PropertyType::Object: return type_Link;
+        default: REALM_TERMINATE("unknown type");
```

Comment:
On that same vein it seems like omitting `Object` may be sensible since the column creation logic for link columns is different than for other column types.